src/terms/nonlocal.jl

```diff
@@ -63,8 +63,8 @@ end
         C = build_projection_coefficients_(T, element.psp)
         for (ik, kpt) in enumerate(basis.kpoints)
             # we compute the forces from the irreductible BZ; they are symmetrized later
-            qs_cart = Gplusk_vectors_cart(basis, kpt)
             qs = Gplusk_vectors(basis, kpt)
+            qs_cart = Array(Gplusk_vectors_cart(basis, kpt))  # Get on the CPU
```

Comment:
and `to_gpu`

Reply:
> maybe on all these comments move them to something like # GPU: move to the CPU so that it's clear these are GPU code paths.

Hmm. It's not just for GPUs (well for now it is ... but this comment would become outdated if we implement other architectures). That's why I removed it.

We really should buy into one of the helper infrastructures Valentin linked above for this `to_cpu` / `to_gpu` stuff before building yet our own.